src/test.rs

```diff
@@ -153,16 +154,17 @@ where
     B: MessageBody + Unpin,
     B::Error: Into<Error>,
 {
-    let mut resp = app
+    let resp = app
         .call(req)
         .await
         .unwrap_or_else(|e| panic!("read_response failed at application call: {}", e));
 
-    let mut body = resp.take_body();
+    let body = resp.into_body();
     let mut bytes = BytesMut::new();
 
-    while let Some(item) = body.next().await {
-        bytes.extend_from_slice(&item.unwrap());
+    actix_rt::pin!(body);
+    while let Some(item) = poll_fn(|cx| body.as_mut().poll_next(cx)).await {
+        bytes.extend_from_slice(&item.map_err(Into::into).unwrap());
```

Comment:
Cant `map_err(Into::into).unwrap()` be replaced with `unwrap_or_else(Into::into)` (here and in a method below)?

Reply:
nah, the panic behavior from before should stay but we need to convert the `Into<Error>` type into `Error` since we're not declaring a Debug bound.